### PR TITLE
Bring master to a release-able state

### PR DESCRIPTION
**Pull request by apoikos**
This is a collection of fixes and optimizations to the build toolchain, as well as some linting to the code, that bring the master branch to a release-able state. In short:

 - Fix all tools under autotools/ to play nice with Python 3
 - Have devel/release run parallel builds by default (optimization, makes building faster)
 - Fix some overlong lines that crept into the code during the Python 3 / GHC 8.6 conversion.

With these changes, `make distcheck-release && make dist-release` succeeds and we are only missing the 3.0.0~beta1 changelog to make a release.

### CODE REVIEW

**Comment by rbott** [comment]
This seems to have broken live migration, I'll look into that later:
```
 Thu May 28 01:05:48 2020 Migrating instance kvm-test-instance02
Thu May 28 01:05:48 2020 * checking disk consistency between source and target
Thu May 28 01:05:49 2020 * closing instance disks on node gnt-test01
Thu May 28 01:05:49 2020 * changing into standalone mode
Thu May 28 01:05:50 2020 * changing disks into dual-master mode
Thu May 28 01:05:51 2020 * wait until resync is done
Thu May 28 01:05:51 2020 * opening instance disks on node gnt-test03 in shared mode
Thu May 28 01:05:52 2020 * opening instance disks on node gnt-test01 in shared mode
Thu May 28 01:05:52 2020 * preparing gnt-test01 to accept the instance
Thu May 28 01:05:52 2020 * migrating instance to gnt-test01
Thu May 28 01:05:53 2020 Migration failed, aborting
Thu May 28 01:05:53 2020 * closing instance disks on node gnt-test01
Thu May 28 01:05:53 2020 * changing into standalone mode
Thu May 28 01:05:53 2020 * changing disks into single-master mode
Thu May 28 01:05:55 2020 * wait until resync is done
Stderr was:
---- FAILED [TestInstanceMigrate] gnt-instance migrate: Command 'gnt-instance migrate --force kvm-test-instance02' on node gnt-test01 failed, exit code 1
>>>> 2020-05-28 01:05:55.701122 time=0:00:08.785230 [TestInstanceMigrate] gnt-instance migrate
---- BLOCK FAILED [RunInstanceTests] Create and exercise instances: Command 'gnt-instance migrate --force kvm-test-instance02' on node gnt-test01 failed, exit code 1
^^^^ BLOCK 2020-05-28 01:05:55.701358 time=0:13:06.014536 [RunInstanceTests] Create and exercise instances
Warning: Permanently added 'gnt-test01,192.168.122.11' (ECDSA) to the list of known hosts.
Failure: command execution error:
Could not migrate instance kvm-test-instance02: Failed to migrate instance: kvm: error executing the migrate-set-capabilities command: Invalid parameter '' (GenericError):
Traceback (most recent call last):
  File "./ganeti-qa.py", line 1135, in <module>
    main()
  File "/usr/share/ganeti/default/ganeti/rapi/client.py", line 274, in wrapper
    return fn(*args, **kwargs)
  File "./ganeti-qa.py", line 1130, in main
    RunQa()
  File "./ganeti-qa.py", line 1024, in RunQa
    RunTestBlock(RunInstanceTests)
  File "./ganeti-qa.py", line 169, in RunTestBlock
    return fn(*args, **kwargs)
  File "./ganeti-qa.py", line 870, in RunInstanceTests
    RunHardwareFailureTests(instance, inodes)
  File "./ganeti-qa.py", line 529, in RunHardwareFailureTests
    RunTestIf("instance-migrate", qa_instance.TestInstanceMigrate, instance)
  File "./ganeti-qa.py", line 150, in RunTestIf
    RunTest(fn, *args, **kwargs)
  File "./ganeti-qa.py", line 112, in RunTest
    retval = fn(*args, **kwargs)
  File "/usr/share/ganeti/testsuite/qa/qa_utils.py", line 819, in wrapper
    result = fn(*args, **kwargs)
  File "/usr/share/ganeti/testsuite/qa/qa_instance.py", line 536, in TestInstanceMigrate
    AssertCommand(cmd)
  File "/usr/share/ganeti/testsuite/qa/qa_utils.py", line 225, in AssertCommand
    _AssertRetCode(rcode, fail, cmdstr, nodename)
  File "/usr/share/ganeti/testsuite/qa/qa_utils.py", line 158, in _AssertRetCode
    (cmdstr, nodename, rcode))
qa.qa_error.Error: Command 'gnt-instance migrate --force kvm-test-instance02' on node gnt-test01 failed, exit code 1
```

This is the local noded log output:
```
2020-05-28 04:08:00,707: ganeti-noded pid=30733 ERROR Failed to migrate instance: kvm: error executing the migrate-set-capabilities command: Invalid parameter '' (GenericError):
Traceback (most recent call last):
  File "/usr/share/ganeti/3.0/ganeti/backend.py", line 3138, in MigrateInstance
    hyper.MigrateInstance(cluster_name, instance, target, live)
  File "/usr/share/ganeti/3.0/ganeti/hypervisor/hv_kvm/__init__.py", line 180, in wrapper
    return fn(self, *args, **kwargs)
  File "/usr/share/ganeti/3.0/ganeti/hypervisor/hv_kvm/__init__.py", line 2566, in MigrateInstance
    self.qmp.SetMigrationCapabilities(migration_caps)
  File "/usr/share/ganeti/3.0/ganeti/hypervisor/hv_kvm/monitor.py", line 224, in wrapper
    ret = fn(*args, **kwargs)
  File "/usr/share/ganeti/3.0/ganeti/hypervisor/hv_kvm/monitor.py", line 733, in SetMigrationCapabilities
    self.Execute("migrate-set-capabilities", arguments)
  File "/usr/share/ganeti/3.0/ganeti/hypervisor/hv_kvm/monitor.py", line 423, in Execute
    ret = self._GetResponse(command)
  File "/usr/share/ganeti/3.0/ganeti/hypervisor/hv_kvm/monitor.py", line 446, in _GetResponse
    raise errors.HypervisorError("kvm: error executing the %s"
ganeti.errors.HypervisorError: kvm: error executing the migrate-set-capabilities command: Invalid parameter '' (GenericError):
```

Details: https://salsa.debian.org/rbott-guest/ganeti-build/-/jobs/770857

File: lib/hypervisor/hv_kvm/__init__.py [hunk 1]
```diff
@@ -2554,12 +2555,15 @@ def MigrateInstance(self, cluster_name, instance, target, live_migration):
...
+    max_bandwidth_in_bytes = \
+        instance.hvparams[constants.HV_MIGRATION_BANDWIDTH] * 1024 * 1024
+    self.qmp.SetMigrationParameters(max_bandwidth_in_bytes,
+        instance.hvparams[constants.HV_MIGRATION_DOWNTIME])
 
     migration_caps = instance.hvparams[constants.HV_KVM_MIGRATION_CAPS]
+    migration_caps = migration_caps.split(_MIGRATION_CAPS_DELIM)
     if migration_caps:
-      self.qmp.SetMigrationCapabilities(migration_caps.split(_MIGRATION_CAPS_DELIM))
+      self.qmp.SetMigrationCapabilities(migration_caps)
```
Comment (saschalucas) [inline lib/hypervisor/hv_kvm/__init__.py:2567]: could the broken live migration be caused by this change? the `split()` was pushed up before the conditional `if migration_caps` . Somehow migration_caps seems to be `''` (according to the error output):
```
ganeti.errors.HypervisorError: kvm: error executing the migrate-set-capabilities command: Invalid parameter '' (GenericError):
```

Comment (apoikos) [reply lib/hypervisor/hv_kvm/__init__.py:2567]: Well spotted, I'll fix that!

Comment (rbott) [reply lib/hypervisor/hv_kvm/__init__.py:2567]: I can confirm that #1472 has fixed the issue, well done :-)

**Pull request merged**